src/components/AnswersProvider/AnswersProvider.test.js

```diff
@@ -51,4 +78,94 @@ describe('QuestionsProvider', () => {
 
     expect(mockedOnClick).toHaveBeenCalledWith(expectedState);
   });
+
+  describe('Batch updates', () => {
+    const customRender = (customProps) =>
+      render(
+        <AnswersProvider {...defaultProps} {...customProps}>
+          <AnswersContext.Consumer>
+            {({ answers, updateAnswers }) => (
+              <>
+                <button
+                  onClick={() =>
+                    updateAnswers({
+                      [Object.keys(answers).length]: { answer: 'DT' },
+                    })
+                  }
+                >
+                  Add answer
+                </button>
+              </>
+            )}
+          </AnswersContext.Consumer>
+        </AnswersProvider>,
+      );
+
+    it('should synchronize answers once the user reaches the limit of 5 new answers', async () => {
+      customRender();
+
+      const addAnswerButton = await screen.findByRole('button', {
+        name: 'Add answer',
+      });
+
+      for (let i = 0; i < 10; i++) {
+        fireEvent.click(addAnswerButton);
```

Comment:
[Small improvement](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event), but this is not needed in this pr

Reply:
You could replace every usage also in this file but this is not needed.